tests/phpunit/integration/tests/Speculative_Prerendering.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+declare(strict_types = 1);
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\Tests\Integration;
+
+use Google\Web_Stories\Admin\Dashboard;
+use Google\Web_Stories\Context;
+use Google\Web_Stories\Story_Post_Type;
+use PHPUnit\Framework\MockObject\MockObject;
+
+/**
+ * @coversDefaultClass \Google\Web_Stories\Speculative_Prerendering
+ */
+class Speculative_Prerendering extends DependencyInjectedTestCase {
+	/**
+	 * @var Context & MockObject
+	 */
+	private $context;
+
+	/**
+	 * @var Story_Post_Type & MockObject
+	 */
+	private $story_post_type;
+
+	/**
+	 * @var Dashboard & MockObject
+	 */
+	private $dashboard;
+
+	protected \Google\Web_Stories\Speculative_Prerendering $instance;
+
+	public function set_up(): void {
+		parent::set_up();
+
+		$this->dashboard       = $this->createMock( Dashboard::class );
+		$this->story_post_type = $this->createMock( Story_Post_Type::class );
```

Comment:
Not necessary to use a mock for this class here. Can just use `$this->injector->make( Story_Post_Type::class )`

Reply:
Can the same be done for the context and dashboard?

Reply:
Also, please expect some more commits on this PR 😅. Anurag and I spent a while trying to get a working local test environment, but it didn't happen. So, we can only run the tests on the PR.

Reply:
Mhm maybe.

If you do it for `Dashboard`, then you'll just have to call `$this->dashboard->add_menu_page()` to the hook suffix exists.

If you do it for `Context`, then you'll have do something like `$GLOBALS['current_screen'] = convert_to_screen( Story_Post_Type::POST_TYPE_SLUG );` to set the current screen, so that the context methods return the correct thing.

Reply:
In that case, I suppose we could do it just for `$story_post_type` for now.

Reply:
Using `$this->injector->make( Story_Post_Type::class )` for `$story_post_type` throws an error here:

https://github.com/GoogleForCreators/web-stories-wp/blob/289baa482e974ff9d870c0b3d88b8bd7580d1733/tests/phpunit/integration/tests/Speculative_Prerendering.php#L96

```
Undefined method 'method'
```

Reply:
Well, `$this->injector->make( Story_Post_Type::class )` creates a proper instance of the class, not a mock.
Without a mock class, you also don't need to mock the `get_slug` method. You can just remove this line.